docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerJNLPLauncher.java

```diff
@@ -161,7 +161,10 @@ public void launch(SlaveComputer computer, TaskListener listener) throws IOExcep
 
     @Override
     public ComputerLauncher getPreparedLauncher(String cloudId, DockerTemplate template, InspectContainerResponse containerInspectResponse) {
-        return new DockerComputerJNLPLauncher(getJnlpLauncher());
+        DockerComputerJNLPLauncher dockerComputerJNLPLauncher = new DockerComputerJNLPLauncher(getJnlpLauncher());
+        DockerComputerJNLPLauncher launcher = (DockerComputerJNLPLauncher) template.getLauncher();
+        dockerComputerJNLPLauncher.setUser(launcher.getUser());
+        return dockerComputerJNLPLauncher;
```

Comment:
It should be implemented in other way, using clone interfaces like SSHLauncher that need to be fixed also. It was a temp hack.